packages/integration-sdk-core/src/types/step.ts

```diff
@@ -11,13 +11,26 @@ import {
 import { IntegrationInstanceConfig } from './instance';
 import { RelationshipDirection } from './relationship';
 
+export enum DisabledStepReason {
+  NONE = 'none', // No reason was provided
+  PERMISSION = 'permission', // Missing permission disabled this step
+  BETA = 'beta', // Step is in beta and only enabled on request
+  CONFIG = 'config', // Step was disabled via config
```

Comment:
Is there any interest in differentiating between a step that was explicitly skipped due to one of the listed reasons and a step that needs to be skipped due to a dependency being skipped?